CMakeLists.txt

```diff
@@ -172,6 +172,9 @@ if(USE_CUDA)
     if(CUDA_VERSION VERSION_GREATER_EQUAL "10.0")
         list(APPEND CUDA_ARCHS "7.5")
     endif()
+    list(JOIN CUDA_ARCHS "\;" CUDA_ARCHS_STRING)
```

Comment:
In example they use a whitespace as a separator.
```
set_property(TARGET tgt PROPERTY CUDA_ARCHITECTURES 35 50 72)
```
I'm not a CMake expert, so not sure whether it matters. Do you feel confident about using `;` here?

My dirty guess is that one can pass any values to this property in CMake 3.20.2 version (which is one we use at our CI) just with the aim to shut up the warning. This is **not** what we want to achieve in this PR.

```
set_property(TARGET tgt PROPERTY CUDA_ARCHITECTURES foo bar)
```
> CMake version 3.20.2 warns when the policy is not set and uses `OLD` behavior.